### ISSUE TITLE
Feature request: expose keyboard button mapping in TIC-80 API

### ISSUE THREAD

**Issue by borbware** [nesbox/TIC-80#1663]
The [issue](https://github.com/nesbox/TIC-80/issues/747) of supporting other keyboard layouts than QWERTY is closed, so TIC-80 seems to support multiple keyboard layouts. However, to my knowledge, the information of the used layout is not currently available in the API.

In a game I'm making, I have a help screen explaining controls. Unfortunately, I can show the correct keyboard button layout only for QWERTY layout users at the moment. If there was a way to know which keyboard keys are the TIC-80 buttons mapped to, I could draw the correct keys on screen.

For example, it could be something like this:

`btnmap(6) -> returns 24 (X key) by default`

![screen1](https://user-images.githubusercontent.com/19384640/140295304-e51b9e34-e131-4fbd-8d11-9d046d7d61b5.gif)

**Comment by nesbox** [comment]
The #747 issue was about text input only (console and code editor).
And I still think keyboard mapping is a part of system configuration and has to be added to the game menu, not to the API.
Also, in the menu could be predefined keyboard layouts.

**Comment by joshgoebel** [comment]
> And I still think keyboard mapping is a part of system configuration and has to be added to the game menu, not to the API.

Agreed, and to be clear no one is requesting it be configured via the API.  

This issue is asking to QUERY the mapping so that the game could show proper instructions to the player such as:

-  "Use the Z key" in the US vs
-  "Use the [whatever] key" in Germany

In order to do that the cartridge would need an API call to learn about it's locale and keyboard configuration... so did you misunderstand or you just don't think this is information the cartridges should ever have access to - and hence displaying directions like what is desired here would simply be impossible?

**Comment by nesbox** [comment]
Heck, it looks like we will have to add an API function that returns the label of the key from its code given the system keyboard mapping.

**Comment by redej0** [comment]
Would it be possible to allow keyboard type?  Currently it is mapped to to Italy.

The mapping is defined here: https://github.com/nesbox/TIC-80/blob/0be2fd8b157284410c373a6b213118a0ba390f28/src/system/baremetalpi/keycodes.h

msx80 wrote:
It's based on the Italian keyboard, the map is derived from here at circle (https://github.com/rsta2/circle/blob/bbc9264ead9dc9e0cf5fc2d4530c26b55ca78c29/lib/input/keymap_it.h). There are other layouts available in this folder, it shouldn't be hard to port over.
Btw circle already has his Keymap system (and kernel option), perhaps we can let circle do the mapping and then use the mapped value, somehow.

**Comment by nesbox** [comment]
It will be fixed here #1613

**Comment by joshgoebel** [comment]
Pretty sure that isn't what this issue is.  This issue wats the mapping exposed via the API so that cartridges can query the mapping to show better help instructions.

**Comment by nesbox** [comment]
Yes, but I'm going to add a keyboard layout switcher to the menu and at the same time API to get the label of the key according to the selected layout.

**Comment by joshgoebel** [comment]
Awesome, but just saying you're "going to add" it doesn't complete/close the issue, does it? :-)   I thought the issues were closed when the work was committed?

**Comment by nesbox** [comment]
Ok, we can reopen all these tasks referenced in #1748 and close later, doesn't matter :)

**Comment by borbware** [comment]
yeah, reopening would be clearer imo. the issues show up as closed in #1748 after all, even tho they're not yet done

**Comment by nesbox** [comment]
Keyboard mapping data added to the RAM (0x14E04)
```
...
| 14E04 | BUTTONS MAPPING   | 32    |
| 14E24 | ... (free)        | 12764 |
+-------+-------------------+-------+
```
There are 4 gamepads each 8 bytes with mapped keys `up down left right a b x y`, 
for example to read the `A` button mapping pls use `peek(0x14E04 + 4)`.

3e7f3f5
